### ISSUE TITLE
Unable to remove refresh_token grant type from OIDC app

### ISSUE THREAD

**Issue by okra-okta** [okta/terraform-provider-okta#1547]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.4.6

### Affected Resource(s)

- okta_app_oauth

### Terraform Configuration Files

```hcl
resource "okta_app_oauth" "spa_client" {
  label                      = "My App"
  type                       = "browser"
  grant_types                = ["authorization_code","refresh_token"]
  redirect_uris              = ["https://localhost:8080/"]
  post_logout_redirect_uris  = ["https://localhost:8080/"]
  response_types             = ["code"]
  token_endpoint_auth_method = "none"
  consent_method             = "REQUIRED"
  auto_key_rotation          = true
}
```

### Debug Output

Error: failed to update OAuth application: the API returned an error: Api validation failed: App Instance. Causes: errorSummary: ''refresh_token'' grant type is required to configure ''refresh_token'' settings.

### Expected Behavior

After creating an OIDC app with `refresh_token` grant type, if I remove it and run `terraform apply` I encounter the error message above.

#### Can this be done in the Admin UI?

Yes

#### Can this be done in the actual API call?

Yes

### Steps to Reproduce

1. Create OIDC app with the code above
2. Remove `refresh_token` from grant_types
3. `terraform apply`
4. Encounter error

**Comment by monde** [comment]
@duytiennguyen-okta has investigated this and has a fix.

**Comment by monde** [comment]
Okta internal reference: https://oktainc.atlassian.net/browse/OKTA-608736

**Comment by duytiennguyen-okta** [comment]
This is fixed in the release [4.0.2](https://github.com/okta/terraform-provider-okta/releases/tag/v4.0.2) from https://github.com/okta/terraform-provider-okta/pull/1586
